io_scene_godot/converters/mesh.py

```diff
@@ -105,270 +67,285 @@ def triangulate_mesh(mesh):
+        base_key = blender_shape_keys.reference_key
+        for index, shape_key in enumerate(blender_shape_keys.key_blocks):
+            if shape_key != base_key:
+                ret.append((index, shape_key))
+        return ret
 
-def extract_shape_keys(blender_shape_keys):
-    """Return a list of (shape_key_index, shape_key_object) each of them
-    is a shape key needs exported"""
-    # base shape key needn't be exported
-    ret = list()
-    base_key = blender_shape_keys.reference_key
-    for index, shape_key in enumerate(blender_shape_keys.key_blocks):
-        if shape_key != base_key:
-            ret.append((index, shape_key))
-    return ret
+    @staticmethod
+    def validate_morph_mesh_modifiers(mesh_object):
+        """Check whether a mesh has modifiers not
+        compatible with shape key"""
+        # this black list is not complete
+        modifiers_not_supported = (
+            bpy.types.SubsurfModifier,
+        )
+        for modifier in mesh_object.modifiers:
+            if isinstance(modifier, modifiers_not_supported):
+                return False
+        return True
 
+    @staticmethod
+    def intialize_surfaces_morph_data(surfaces):
+        """Initialize a list of empty morph for surfaces"""
+        surfaces_morph_list = [VerticesArrays() for _ in range(len(surfaces))]
 
-def intialize_surfaces_morph_data(surfaces):
-    """Initialize a list of empty morph for surfaces"""
-    surfaces_morph_list = [VerticesArrays() for _ in range(len(surfaces))]
+        for index, morph in enumerate(surfaces_morph_list):
+            morph.vertices = [None] * len(surfaces[index].vertex_data.vertices)
 
-    for index, morph in enumerate(surfaces_morph_list):
-        morph.vertices = [None] * len(surfaces[index].vertex_data.vertices)
+        return surfaces_morph_list
 
-    return surfaces_morph_list
+    def export_morphs(self, export_settings, surfaces):
+        """Export shape keys in mesh node and append them to surfaces"""
+        if export_settings['use_mesh_modifiers']:
+            if not MeshResourceExporter.validate_morph_mesh_modifiers(
+                    self.object):
+                raise ValidationError(
+                    "Mesh object '{}' has modifiers "
+                    "incompatible with shape key".format(self.object.name)
+                )
 
+        self.mesh_resource["blend_shape/names"] = Array(
+            prefix="PoolStringArray(", suffix=')'
+        )
+        self.mesh_resource["blend_shape/mode"] = 0
 
-def validate_morph_mesh_modifiers(mesh_object):
-    """Check whether a mesh has modifiers not
-    compatible with shape key"""
-    # this black list is not complete
-    modifiers_not_supported = (
-        bpy.types.SubsurfModifier,
-    )
-    for modifier in mesh_object.modifiers:
-        if isinstance(modifier, modifiers_not_supported):
-            return False
-    return True
+        shape_keys_to_export = MeshResourceExporter.extract_shape_keys(
```

Comment:
You can use `self.extract_shape_keys` which means changing the class name won't break things.